driver/gather_driver.hpp

```diff
@@ -0,0 +1,389 @@
+/*******************************************************************************
```

Comment:
I noticed that the collection of Gather operations you implemented includes many similar files. In the future, pls consider consolidating all these Gather operations into the same branch to avoid duplicating work.

Reply:
I separated branch to avoid too many details for reviewing, of course in future I would merge those into one.